src/controllers/main/main.ts

```diff
@@ -333,6 +344,98 @@ export class MainController extends EventEmitter {
     this.emitUpdate()
   }
 
+  async importSmartAccountFromDefaultSeed(seed?: string) {
+    await this.withStatus(
+      'importSmartAccountFromDefaultSeed',
+      async () => {
+        if (this.accountAdder.isInitialized) this.accountAdder.reset()
+        if (seed && !this.keystore.hasKeystoreDefaultSeed) {
+          await this.keystore.addSeed(seed)
+        }
+
+        const defaultSeed = await this.keystore.getDefaultSeed()
+
+        if (!defaultSeed) {
+          throw new EmittableError({
+            message:
+              'Failed to retrieve default seed phrase from keystore. Please try again or contact Ambire support if the issue persists.',
+            level: 'major',
+            error: new Error('failed to retrieve default seed phrase from keystore')
+          })
+        }
+
+        const keyIterator = new KeyIterator(defaultSeed)
+        this.accountAdder.init({
+          keyIterator,
+          hdPathTemplate: BIP44_STANDARD_DERIVATION_TEMPLATE,
+          pageSize: 1
+        })
```

Comment:
Works just fine, but I had 2 scenarios with importing seeds that caused this error:

<img width="620" alt="Screenshot 2024-08-05 at 17 23 51" src="https://github.com/user-attachments/assets/c5a1b021-4bfb-466f-9a58-26ed33ccd613">

and an error that we failed to check if the acc is used on Gnosis network.

Best would be to introduce flags if the AccountAdder should: 1) Search for linked accounts and 2) Should check if account is used on the networks. Both should be false for this use-case. Can be moved in another PR, I'm opening an issue.

Reply:
Ref: https://github.com/AmbireTech/ambire-app/issues/2615

Reply:
Fixed in https://github.com/AmbireTech/ambire-common/pull/922